### PR TITLE
[4.2] OGM-693 Queries on embedded collections

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-693

This patch should allow the execution of embedded queries on all the supported datastores but it requires the lates HQL parser and I'm not sure if it has been release yet

### CODE REVIEW

**Comment by Sanne** [comment]
Jenkins, test this again please?

**Comment by Sanne** [comment]
test this please

**Comment by Sanne** [comment]
test this please

**Comment by gunnarmorling** [comment]
@DavideD, could you add a commit for adding the new feature to the list of supported query constructs in the affected datastore chapters?

**Comment by DavideD** [comment]
> @DavideD, could you add a commit for adding the new feature to the list of supported query constructs in the affected datastore chapters?

Right, forgot abut this, by the way I just discovered that projections on properties of the collection don't work well. Trying to fix it.

File: core/src/main/java/org/hibernate/ogm/query/parsing/impl/ParserPropertyHelper.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+			}
+			else if ( subType.isAssociationType() ) {
+				throw new UnsupportedOperationException( "Queries on collection in embeddables are not supported: " + property );
+			}
+			columnNameBuilder.append( "." );
+			columnNameBuilder.append( persister.toColumns( property )[0] );
+		}
+	}
+
+	protected String join(Iterable<String> propertyName) {
```
Comment (gunnarmorling) [inline core/src/main/java/org/hibernate/ogm/query/parsing/impl/ParserPropertyHelper.java]: There is a method for this on `StringHelper` already.

**Comment by Sanne** [comment]
test this please

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-ogm/pull/513

**Pull request closed without merging**